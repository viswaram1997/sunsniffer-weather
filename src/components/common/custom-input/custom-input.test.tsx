import { render, fireEvent } from "@testing-library/react";
import Input from "./index"
const myMock = jest.fn();

describe('Custom input', () => {

    test('should match snapshot', () => {
        const result = render(<Input
            name="city "
            onChange={myMock}
            label='Submit'
            value=""
            placeHolder="test"
        />)
        expect(result.baseElement).toMatchSnapshot();
    })

    test('should display proper value', async () => {
        const result = render(<Input
            name="city "
            onChange={myMock}
            label='Submit'
            value=""
            placeHolder="test"
        />)
        const input = await result.findByPlaceholderText('test')
        expect(input).not.toBeNull()
    })

    test('should call on click function', async () => {
        const result = render(<Input
            name="city "
            onChange={myMock}
            label='Submit'
            value=""
            placeHolder="test"
        />)
        const input = await result.findByPlaceholderText('test')
        fireEvent.change(input, {target: {value: 'test'}})
        expect(myMock).toBeCalled()
    })
    test('should display proper value', async () => {
        const result = render(<Input
            name="city "
            onChange={myMock}
            label='Submit'
            value="test123"
            placeHolder="test"
        />)
        const input:any = await result.findByPlaceholderText('test')
        expect(input.value).toEqual('test123')
    })


});