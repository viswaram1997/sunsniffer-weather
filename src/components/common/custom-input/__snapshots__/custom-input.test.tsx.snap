// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom input should match snapshot 1`] = `
<body>
  <div>
    <div
      class="mb-3 "
    >
      <label
        class="form-label"
      >
        Submit
      </label>
      <input
        class="form-control"
        name="city "
        placeholder="test"
        type="text"
        value=""
      />
      <div
        class="invalid-feedback"
      >
        Please enter the 
        city 
      </div>
    </div>
  </div>
</body>
`;
